docs/user_guide.rst

```diff
@@ -469,35 +469,33 @@ Then, to install from local only, you'll be using :ref:`--find-links
 $ pip install --no-index --find-links=DIR -r requirements.txt
 
 
-"Only if needed" Recursive Upgrade
-**********************************
+Upgrade Behaviour
+*****************
```

Comment:
This whole section could possibly be deleted, since it only existed to give folks a workaround for the old confusing behavior, and after this pr is merged then pip will join most other package managers in not even having a concept of a "eager upgrade". We don't need to emphasize to people that it doesn't have eager upgrade, any more than we need to emphasize that it doesn't have four-wheel drive or email reading capabilities.

Alternatively, since people will often consult the latest manual even for old versions, and I think there are pages out there that link to this section, it might make sense to keep the old information and section title (since the section title determines the link anchor) but reframed as: "it used to be that pip did a weird thing with upgrades, and this section was here to tell you how to work around it. Starting with version X, it doesn't anymore. So the recommended way to work around it is to upgrade to version X or better. As a historical note, the way you used to work around this was by running `pip ...`."